mx.c

```diff
@@ -1685,6 +1685,92 @@ struct Mailbox *mx_path_resolve(const char *path)
   return m;
 }
 
+/**
+ * mx_mbox_find_by_name - Find a Mailbox with given name under an account.
+ * @param a    Account to search
+ * @param name Name to find
+ * @retval ptr Mailbox
+ */
+static struct Mailbox *mx_mbox_find_by_name_ac(struct Account *a, const char *name)
+{
+  if (!a || !name)
+    return NULL;
+
+  struct MailboxNode *np = NULL;
+
+  STAILQ_FOREACH(np, &a->mailboxes, entries)
+  {
+    if (np->mailbox->name && mutt_str_equal(np->mailbox->name, name))
+      return np->mailbox;
+  }
+
+  if (!np)
+    return NULL;
```

Comment:
I'm not sure you can have `np != NULL` here

Reply:
Good point, thanks